lib/client_summarizer.rb

```diff
@@ -0,0 +1,31 @@
+class ClientSummarizer
+  include FiscalYearMixin
+
+  attr_reader :client_namespace, :fiscal_year
+
+  def initialize(args)
+    @client_namespace = args[:client_namespace]
+    now = Time.zone.now
+    @fiscal_year = (args[:fiscal_year] || self.class.which_fiscal_year(now.year, now.month)).to_i
+    @_records = build_records
```

Comment:
since we are using this below, I'd probs name it `@records` -- I've been using underscored ivars to indicate that an ivar is for memoization only (eg: your use in `run`)